js/data/odata/store.js

```diff
@@ -218,54 +205,48 @@ const ODataStore = Store.inherit({
         return query(url, queryOptions);
     },
 
-    _insertImpl: function(values) {
+    _insertImpl(values) {
         this._requireKey();
 
-        const that = this;
         const d = new Deferred();
 
         when(this._sendRequest(this._url, 'POST', null, values))
-            .done(function(serverResponse) {
-                d.resolve(config().useLegacyStoreResult ? values : (serverResponse || values), that.keyOf(serverResponse));
-            })
+            .done(serverResponse =>
+                d.resolve(
+                    serverResponse && !config().useLegacyStoreResult ? serverResponse : values,
+                    this.keyOf(serverResponse)
+                )
+            )
             .fail(d.reject);
 
         return d.promise();
     },
 
-    _updateImpl: function(key, values) {
+    _updateImpl(key, values) {
         const d = new Deferred();
 
-        when(
-            this._sendRequest(this._byKeyUrl(key), this._updateMethod, null, values)
-        ).done(
-            function(serverResponse) {
-                if(config().useLegacyStoreResult) {
-                    d.resolve(key, values);
-                } else {
-                    d.resolve(serverResponse || values, key);
-                }
-            }
-        ).fail(d.reject);
+        when(this._sendRequest(this._byKeyUrl(key), this._updateMethod, null, values))
+            .done(serverResponse =>
+                config().useLegacyStoreResult
+                    ? d.resolve(key, values)
+                    : d.resolve(serverResponse || values, key)
+            )
+            .fail(d.reject);
 
         return d.promise();
     },
 
-    _removeImpl: function(key) {
+    _removeImpl(key) {
         const d = new Deferred();
```

Comment:
```javascript
return this._sendRequest(this._byKeyUrl(key), 'DELETE');
```
?

Reply:
We can't get rid of this construction. The `_sendRequest` return `Deffered` object but on success we return
```js
() => d.reslove(key)
```
We don't use result of `_sendRequest` on success (but use it on fail)